src/cmd/flash.rs

```diff
@@ -55,62 +59,67 @@ pub fn run(args: &Args) -> Result<()> {
     // repo path is needed since cros flash outside chroot only works within the cros checkout
     let repo = &get_repo_dir(&args.repo)?;
 
-    // Determine a BOARD to flash
-    let board = match (&args.board, &args.dut) {
-        (Some(board), None) => board.clone(),
-        (_, Some(dut)) => {
-            ensure_testing_rsa_is_there()?;
-            let dut = &DutInfo::new(dut)?;
-            let board_from_dut = dut
-                .info()
-                .get("board")
-                .context("Failed to get --board from ")?
-                .clone();
-            if let Some(board_from_arg) = &args.board {
-                // The board name may have suffix '64' or '-*', so get the first alphabet
-                // sequence as the base board name.
-                let re = Regex::new(r"(^[[:alpha:]]*)")?;
-                let cap_arg = if let Some(cap) = re.captures(board_from_arg) {
-                    cap
+    let image_path = if let Some(image) = &args.image {
+        // If --image is specified, use the local file
+        image.clone()
+    } else {
+        // Determine a BOARD to flash
```

Comment:
Can you make this `else` block a local function?

Reply:
The part is extracted into determine_board_to_flash(). Please review again!

Reply:
I'm curious if the original comment meant  the whole thing in the else block or the part that was extracted as `determine_board_to_flash`

Reply:
Not sure what do you mean... @mhiramat Could you checkif the change is acceptable and aligned with your suggestion? Thanks!!!

Reply:
ping...

Reply:
Sorry for later. Yes, @HidenoriKobayashi is right. The else block seems independently works.